modules/cactus-web/src/Header/Header.tsx

```diff
@@ -0,0 +1,112 @@
+import PropTypes from 'prop-types'
+import React, { Children, ReactNode } from 'react'
+import styled from 'styled-components'
+import { margin, MarginProps } from 'styled-system'
+
+import Text from '../Text/Text'
+
+export type BackgroundColorVariants = 'lightContrast' | 'white'
+interface HeaderProps extends MarginProps, React.HTMLAttributes<HTMLDivElement> {
+  bgColor?: BackgroundColorVariants
+  children?: ReactNode
+}
+
+export type HeaderType = React.FC<HeaderProps> & {
+  Item: typeof HeaderItem
+  BreadcrumbRow: typeof HeaderBreadcrumbRow
+  Title: typeof HeaderTitle
+}
+
+export const HeaderItem: React.FC<React.HTMLAttributes<HTMLDivElement>> = ({ children }) => {
+  return <>{children}</>
+}
+
+export const HeaderTitle: React.FC = ({ children }) => {
+  return (
+    <Text as="h1" textStyle="h2" my="0">
+      {children}
+    </Text>
+  )
+}
+
+export const HeaderBreadcrumbRow: React.FC<React.HTMLAttributes<HTMLDivElement>> = ({
+  children,
+}) => {
+  return <>{children}</>
+}
+
+export const Header: HeaderType = ({ children, bgColor = 'lightContrast', ...rest }) => {
+  const childrens = Children.toArray(children)
+
+  return (
+    <StyledHeader bgColor={bgColor} {...rest}>
+      <HeaderColumn>
+        {childrens.find(
+          (child: any) => child.type.displayName === 'HeaderBreadcrumbRow' && <>{child}</>
```

Comment:
To avoid the `any` here, you could do the following if you want.  First try adding the following two lines above line 41:

```typescript
type ChildType = React.ReactChild | React.ReactFragment | React.ReactPortal
type ChildElement = React.ReactElement<any, React.ComponentClass | React.FunctionComponent>
```

Then, you can change this line to the following:

```suggestion
          (child: ChildType) =>
            (child as ChildElement).type.displayName === 'HeaderBreadcrumbRow' && <>{child}</>
```

Reply:
Thanks, @mikesoltow ! I wonder if you know about some useful links (tutorials, blogs, etc...) where I could take a deep dive into Typescript and React Types (In addition to the documentation)

Reply:
I don't unfortunately.  I was playing around with this just using VSCode's IntelliSense.  I deleted the `any`s and looked at what it was complaining about.  It indicated that `child` was of type `React.ReactNode`.  I could then CMD+Click on the React types in VS code to be taken to the type definitions.  And I could see that the problem was that `React.ReactNodes` are not guaranteed to have a `displayName` on them (only some descendants of `ReactNode` have that property).  We need to make a [type assertion](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#type-assertions) to tell TypeScript that we know the children will be an element that has a `displayName`.  Some more clicking in the type definition files led me to to the type we needed to assert on.